impeller/compiler/shader_lib/impeller/dithering.glsl

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef DITHERING_GLSL_
+#define DITHERING_GLSL_
+
+#include <impeller/types.glsl>
+
+/// The dithering rate, which is 1.0 / 64.0, or 0.015625.
+const float kDitherRate = 1.0 / 64.0;
+
+/// Returns the closest color to the input color using 8x8 ordered dithering.
+///
+/// Ordered dithering divides the output into a grid of cells, and then assigns
+/// a different threshold to each cell. The threshold is used to determine if
+/// the color should be rounded up (white) or down (black).
+//
+/// This technique was chosen mostly because Skia also uses it:
+/// https://github.com/google/skia/blob/f9de059517a6f58951510fc7af0cba21e13dd1a8/src/opts/SkRasterPipeline_opts.h#L1717
+///
+/// See also:
+/// - https://en.wikipedia.org/wiki/Ordered_dithering
+/// - https://surma.dev/things/ditherpunk/
+/// - https://shader-tutorial.dev/advanced/color-banding-dithering/
+vec4 IPOrderedDither8x8(vec4 color, vec2 dest) {
+  // Get the x and y coordinates of the pixel in the 8x8 grid.
+  uint iota[] = uint[](0, 1, 2, 3, 4, 5, 6, 7);
```

Comment:
be careful declaring stack allocated arrays. I think spirvcross might be hoisting this to a constant, but just in case I would make it one explicitly (and give it a size).

Reply:
Thanks! I'm actually replacing this with simpler code ... I'm guessing the Skia implementation was an optimization specifically for C.

PTAL.